test/Microsoft.DocAsCode.Build.TableOfContents.Tests/TocRestructureTest.cs

```diff
@@ -0,0 +1,331 @@
+            var expected = GetTocItem(@"
+root
+    node1
+        leaf1
+        node2
+        node3
+        node2
+            leaf2.3
+            leaf2.4
+            leaf2
+            leaf3
+            leaf2.1
+            leaf2.2
+        leaf4
+        leaf5
+    leaf6
+");
+            AssertTocEqual(expected, toc);
+        }
+
+        [Fact]
+        public void TestTocRestructureWithNoMatchNode()
+        {
+            var layout = @"
+root
+    node1
+        leaf1
+        node2
+            leaf2
+            leaf3
+    leaf3
+";
+            var toc = GetTocItem(layout);
+            var restructures = new List<TreeItemRestructure>
+            {
+                GetRestructure(TreeItemActionType.AppendChild, "leaf100", new string[] {"leaf2.1", "leaf2.2" }),
+                GetRestructure(TreeItemActionType.PrependChild, "leaf100", new string[] {"leaf2.1", "leaf2.2" }),
+                GetRestructure(TreeItemActionType.InsertAfter, "leaf100", new string[] {"leaf4", "leaf5" }),
+                GetRestructure(TreeItemActionType.InsertBefore, "leaf100", new string[] {"node2", "node3" }),
+                GetRestructure(TreeItemActionType.DeleteSelf, "leaf100", new string[] {"leaf6", "leaf7" }),
+                GetRestructure(TreeItemActionType.ReplaceSelf, "leaf100", new string[] {"leaf6", "leaf7" }),
+            };
+            tocBuilder.Restructure(toc, restructures);
+            var expected = GetTocItem(layout);
+            AssertTocEqual(expected, toc);
+        }
+
+        [Fact]
+        public void TestReplaceNodeWithMultipleNodesThrows()
+        {
+            var toc = GetTocItem(@"
+root
+    node1
+        leaf1
+        leaf2
+    leaf3
+    node2
+        node1
+");
+            var restructures = new List<TreeItemRestructure>
+            {
+                GetRestructure(TreeItemActionType.ReplaceSelf, "node2", new string[] {"leaf4", "leaf5" }),
+            };
+            Assert.Throws<InvalidOperationException>(() => tocBuilder.Restructure(toc, restructures));
+        }
+
+        [Fact]
+        public void TestTocRestructureWithRestructureConflicts()
+        {
+            var toc = GetTocItem(@"
+root
+    node1
+        leaf1
+        leaf2
+    leaf3
+    node2
+        node1
+");
+            var restructures = new List<TreeItemRestructure>
+            {
+                GetRestructure(TreeItemActionType.InsertAfter, "node2", new string[] {"leaf4", "leaf5" }),
+                GetRestructure(TreeItemActionType.DeleteSelf, "node2", new string[] {"leaf6", "leaf7" }),
+                GetRestructure(TreeItemActionType.ReplaceSelf, "node2", new string[] {"leaf8", "leaf9" }),
+                GetRestructure(TreeItemActionType.InsertBefore, "node2", new string[] {"leaf10", "leaf11" }),
+            };
+            tocBuilder.Restructure(toc, restructures);
+            var expected = GetTocItem(@"
+root
+    node1
+        leaf1
+        leaf2
+    leaf3
+    node2
```

Comment:
should be deleted, test failure.

---
In reply to: [97012531](https://github.com/dotnet/docfx/pull/1204#discussion_r97012531) [](ancestors = 97012531)